Source/Diagnostics/ReducedDiags/ParticleHistogram.cpp

```diff
@@ -211,18 +211,19 @@ void ParticleHistogram::ComputeDiags (int step)
                 {
                     amrex::ParticleReal x, y, z;
                     GetPosition(i, x, y, z);
-                    auto const w  = d_w[i];
+                    auto const w  = (amrex::Real)d_w[i];
```

Comment:
For all the things below, I would rather stay/cast to `ParticleReal`. But is it possible that all the functor stuff accepts only `Real`?

Reply:
@PhilMiller I am concerned about the cost of a cast on GPU, it's a special instruction and "surprisingly" costly.

Reply:
Today I learned. Thanks

Reply:
You should probably leave it as is, to emphasize that a cast operation is occurring, which may be a noticeable expense.